### ISSUE TITLE
bug: oil breaks netrw

### ISSUE THREAD

**Issue by zim0369** [stevearc/oil.nvim#148]
### Did you check the docs and existing issues?

- [X] I have read the docs
- [X] I have searched the existing issues

### Neovim version (nvim -v)

NVIM v0.9.1 Build type: Release LuaJIT 2.1.0-beta3     system vimrc file: "$VIM/sysinit.vim"   fall-back for $VIM: "/usr/share/nvim"  Run :checkhealth for more info

### Operating system/version

Linux arch 6.4.4-arch1-1 #1 SMP PREEMPT_DYNAMIC Wed, 19 Jul 2023 19:19:38 +0000 x86_64 GNU/Linux

### Describe the bug

After installing oil.nvim and then opening netrw using `Ex` everything looks just fine but when reopening it or `sorting` using `s`, netrw is unable to show anything.

### Steps To Reproduce

1. open `nvim`
2. enter cmd `:Ex`
3. close `netrw`
4. reopen netrw. No files visible
5. or toggle sort in netrw using `s`
6. again no files or dirs are visible

### Expected Behavior

Netrw should function as normal

### Repro

```Lua
-
```

### Did you check the bug with a clean config?

- [X] I have confirmed that the bug reproduces with `nvim -u repro.lua` using the repro.lua file above.

**Comment by zim0369** [comment]
Opened this issues because sometimes I'd like to use netrw as well.

**Comment by stevearc** [comment]
Can you provide a minimal repro file so I can test this locally?

**Comment by zim0369** [comment]
Opening `nvim` with `nvim -u ~/oil.lua`, where contents of the file `~/oil.lua` are:
```lua
vim.cmd 'set rtp+=/home/zim/.local/share/nvim/lazy/oil.nvim'
require("oil").setup()
```
Then we open netrw using `:Explore`:
![image](https://github.com/stevearc/oil.nvim/assets/85027668/11e99109-74a8-448f-9e02-0bd321da25a5)
Everything looks fine except for the syntax highlight
Now we close netrw using `:bd` & relaunch netrw using `:Ex`:
![image](https://github.com/stevearc/oil.nvim/assets/85027668/cbacb72e-2ae4-4cf6-ba07-c9f70cdd52a9)
Now no files or dirs are listed. This doesn't seem like a highlight problem because I'm unable to scroll below.

Here's how netrw looks like without `oil.nvim` installed:
![image](https://github.com/stevearc/oil.nvim/assets/85027668/a1985862-4d56-49d1-9381-41f3a23f6c03)
and is able to open like this in subsequent launches.

There's one more problem which I forgot to reproduce with a minimal config. What happens is if I open any file, then open netrw, then open oil from netrw & close oil, I end up in a blank buffer which is actually a non-existent buffer and is not listed in `buffers`. So to navigate to back to that file I have to switch to that buffer explicitly. This is how it looks:
![image](https://github.com/stevearc/oil.nvim/assets/85027668/6a85ef18-6bb9-4959-9b7e-72c2f1e81792)

Notice how the only listed buffers in the `:buffers` output are the files that are open and not the current blank buffer left after closing `oil`.

**Comment by zim0369** [comment]
Also, it'd be convenient if launching `oil` from a `netrw` buffer launched `oil` in the same parent dir. Right now I'm using this as a workaround:
```lua
    vim.keymap.set("n", "<leader>e", function()
      local path = vim.b.netrw_curdir
      vim.cmd({ cmd = "Oil", args = { path } })
    end, { desc = "close oil", buffer = true })
```

**Comment by stevearc** [comment]
This is because Oil will try to be the default file explorer and take over buffers that have the name of a directory. If you don't want that behavior, set `default_file_explorer = false` in `setup()`. I've added some clearer messaging to the notification that shows up in netrw buffers.

I've also moved your workaround into the plugin, so now opening oil from a netrw buffer should default to opening in the same location.
